lib/active_reporting/reporting_dimension.rb

```diff
@@ -10,43 +10,52 @@ class ReportingDimension
     DATETIME_HIERARCHIES = %i[microseconds milliseconds second minute hour day week month quarter year decade
                               century millennium].freeze
     JOIN_METHODS = { joins: :joins, left_outer_joins: :left_outer_joins }.freeze
-    attr_reader :join_method
+    attr_reader :join_method, :label
 
-    def_delegators :@dimension, :name, :type, :klass, :association, :model, :hierarchical?, :datetime?
+    def_delegators :@dimension, :name, :type, :klass, :association, :model, :hierarchical?
 
     def self.build_from_dimensions(fact_model, dimensions)
       Array(dimensions).map do |dim|
-        dimension_name, label = dim.is_a?(Hash) ? Array(dim).flatten : [dim, nil]
+        dimension_name, options = dim.is_a?(Hash) ? Array(dim).flatten : [dim, nil]
         found_dimension = fact_model.dimensions[dimension_name.to_sym]
 
-        raise(UnknownDimension, "Dimension '#{dim}' not found on fact model '#{fact_model}'") if found_dimension.nil?
+        raise(UnknownDimension, "Dimension '#{dimension_name}' not found on fact model '#{fact_model}'") if found_dimension.nil?
 
-        new(found_dimension, **label_config(label))
+        # Ambiguous behavior with string option for degenerate and standard dimension
+        if !options.is_a?(Hash) && found_dimension.type == Dimension::TYPES[:degenerate]
+          warn "[DEPRECATION] direct use of implict hierarchies is deprecated. Please use `:datetime_drill` option instead."
```

Comment:
To be honest, I've never written any libraries where I've deprecated and eventually removed something. Should the message mentioned that this old way of doing things will be removed in a future version?

Reply:
You are right. I changed the string and use the right method `ActiveSupport::Deprecation.warn`